pkg/util/fuzzystrmatch/leven_test.go

```diff
@@ -206,3 +206,139 @@ func TestLevenshteinDistanceWithCost(t *testing.T) {
 		}
 	}
 }
+
+func TestLevenshteinDistanceWithNegativeCost(t *testing.T) {
+	
+	tc:= struct {
+		Source   string
+		Target   string
+		Expected int
+	}{
+		Source:   "book",
+		Target:   "back",
+		Expected: 8,
+	}
+}
+
+func TestLevenshteinDistanceWithCostAndThreshold(t *testing.T) {
+	tt := []struct {
+		Source   string
+		Target   string
+		InsertionCost int
+		DeletionCost int
+		SubstitutionCost int
+		Threshold int
+		Expected int
+	}{
+		{
+			Source:   "book", 
+			Target:   "back", 
+			InsertionCost: 1, 
+			DeletionCost: 1, 
+			SubstitutionCost: 1, 
+			Threshold: 8, 
+			Expected: 2,
+		},
+		{
+			Source:   "vacaville", 
+			Target:   "fairfield", 
+			InsertionCost: 1, 
+			DeletionCost: 2, 
+			SubstitutionCost: 1, 
+			Threshold: 24, 
+			Expected: 6,
+		},
+		{
+			Source:   " ", 
+			Target:   "123456789", 
+			InsertionCost: 1, 
+			DeletionCost: 1, 
+			SubstitutionCost: 1, 
+			Threshold: 9, 
+			Expected: 9,
+		},
+		{
+			Source:   "123456789", 
+			Target:   " ", 
+			InsertionCost: 4, 
+			DeletionCost: 2, 
+			SubstitutionCost: 1, 
+			Threshold: 27, 
+			Expected: 17,
+		},
+		{
+			Source:   "123456789", 
+			Target:   "123", 
+			InsertionCost: 1, 
+			DeletionCost: 1, 
+			SubstitutionCost: 1, 
+			Threshold: 18, 
+			Expected: 6,
+		},
+		{		
+			Source:   " ", 
+			Target:   "123456789", 
+			InsertionCost: 2, 
+			DeletionCost: 3, 
+			SubstitutionCost: 4, 
+			Threshold: 18, 
+			Expected: 19,
+		},
+		{	
+			Source:   "123456789", 
+			Target:   " ", 
+			InsertionCost: 5, 
+			DeletionCost: 6, 
+			SubstitutionCost: 7, 
+			Threshold: 27, 
+			Expected: 28,
+		},
+		{
+			Source:   "alfaromeo",
+			Target:   "alfaromeoaudibenz",
+			InsertionCost: 1, 
+			DeletionCost: 1, 
+			SubstitutionCost: 1, 
+			Threshold: 19,
+			Expected: 8,
+		},
+		{
+			Source:   "alfaromeo",
+			Target:   "bayerischemotorenwerkealfaromeoaudibenz",
+			InsertionCost: 1, 
+			DeletionCost: 1, 
+			SubstitutionCost: 1, 
+			Threshold: 30,
+			Expected: 31,
```

Comment:
i see in postgres that the result is 30:
```
postgres=# select levenshtein_less_equal('alfaromeo', 'bayerischemotorenwerkealfaromeoaudibenz', 1,1,1, 30);
 levenshtein_less_equal
------------------------
                     30
```

Reply:
Fixed this by removing early exit logic.